staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go

```diff
@@ -156,6 +156,44 @@ func testCheckResult(t *testing.T, expectEventType watch.EventType, w watch.Inte
 	})
 }
 
+func checkDiff(t *testing.T, expected, got interface{}) bool {
+	t.Helper()
+	if !reflect.DeepEqual(expected, got) {
+		if diff := cmp.Diff(expected, got); diff != "" {
+			return true
+		}
+		return true
+	}
+	return false
+}
+
+func testCheckResultForScopedWatch(t *testing.T, expectEventType watch.EventType, w watch.Interface, expectObj *example.Pod) {
+	passTest, err := testCheckResultFuncForScopedWatch(t, expectEventType, w, func(object runtime.Object) bool {
+		return checkDiff(t, expectObj, object)
+	})
+	if !passTest {
+		t.Errorf("test failed with error: %v", err)
+	}
+}
+
+func testCheckResultFuncForScopedWatch(t *testing.T, expectEventType watch.EventType, w watch.Interface, check func(object runtime.Object) bool) (bool, error) {
+	for {
```

Comment:
Can you clarify why do you need the loop?

Given that we control all the load (we create/update/delete all objects), we should know exactly what events we expect. Why do you need this?

Reply:
Thank you for your suggestion! I added the list operation and used the resource version of the list as a start to watch the object. Then I don't need the foo loop now.